### PR TITLE
chore: separate npcheck runs to their own workflow

### PR DESCRIPTION
**Pull request by richardlau**
Run `npcheck` in its own workflow rather than part of the documentation
StyleCheck workflow.

### CODE REVIEW

**Comment by richardlau** [comment]
Well https://github.com/nodeshift/nodejs-reference-architecture/runs/4290103480?check_suite_focus=true ran but fails. It's hard to see which are the two errors, although https://github.com/nodeshift/nodejs-reference-architecture/pull/98 should fix the one about `nyc`'s license.

**Review by helio-frota — APPROVED** [review]

**Comment by helio-frota** [comment]
> It's hard to see which are the two errors. 

I agree, I saw the other error is about license as well.
another day, I added the raw result in a txt file and started to delete other things until end up with 2 license errors

**Comment by richardlau** [comment]
Ah using the UI to search for "FAIL" gives:
https://github.com/nodeshift/nodejs-reference-architecture/runs/4290103480?check_suite_focus=true#step:4:69
```
Checking license of argparse@2.0.1                                          FAIL 
```
and
https://github.com/nodeshift/nodejs-reference-architecture/runs/4290103480?check_suite_focus=true#step:4:1740
```
Checking license of caniuse-lite@1.0.30001282                               FAIL 
```
corresponding to
```
(5): The module "eslint" depends on the "argparse@2.0.1" package which is under the non-acceptable license "Python-2.0".
...
(44): The module "nyc" depends on the "caniuse-lite@1.0.30001282" package which is under the non-acceptable license "CC-BY-4.0".
```

As previously mentioned, https://github.com/nodeshift/nodejs-reference-architecture/pull/98 should fix the `caniuse-lite` license error.

File: .github/workflows/npcheck.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# This workflow will run the due dilligence checks provided by `npcheck`:
+# - On demand.
+# - Regularly (once a week).
+# - When `npcheck.json` or this workflow file is changed.
+
```
Comment (mhdawson) [inline .github/workflows/npcheck.yml:5]:
```suggestion


```

File: .github/workflows/npcheck.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
```
Comment (mhdawson) [inline .github/workflows/npcheck.yml:5]:
```suggestion
```

**Comment by mhdawson** [comment]
Ok, fixed up the npcheck file and allowed Python-2 just for eslint as it `seems` ok to me.

**Review by mhdawson — APPROVED** [review]
LGTM

**Pull request merged**